### ISSUE TITLE
Netbird server Helm chart

### ISSUE THREAD

**Issue by MohammedNoureldin** [netbirdio/netbird#853]
Hi,

I have created this issue to track the status of creating a Helm chart to install a self-hosted server easily. I have just finished preparing all Kubernetes required resources and got it working out of the box.

I just still need to replace the hard-coded fields with Helm vars.

Here I am using no "magic script" to create my configuration, because it was the most tedious part for me to get everything working without the script, because it reduced the readability and my understanding to what is going on and how is every part being configured. So I think these K8s resources can also help anybody to understand how the components can be configured manually.

**Comment by jayvardhan2111** [comment]
You can assign me this task.
Also, Please help me in finding Kubernetes manifest files so that I can start working on packaging it using Helm.

**Comment by M0NsTeRRR** [comment]
Hello,
Any news on this feature request ?
Regards,

**Comment by M0NsTeRRR** [comment]
Hello @MohammedNoureldin @jayvardhan2111 
Any news on this feature request ?
Regards,

**Comment by M0NsTeRRR** [comment]
As I see jaconi company have shared helm chart for client and server https://artifacthub.io/packages/helm/jaconi/netbird 

https://github.com/jaconi-io/helm-charts/tree/main

**Comment by dextercai** [comment]
HelmCharts from jaconi's company seems a bit complicated, I think there can be a simplified version for easier understanding.

**Comment by dextercai** [comment]
> As I see jaconi company have shared helm chart for client and server https://artifacthub.io/packages/helm/jaconi/netbird
> 
> https://github.com/jaconi-io/helm-charts/tree/main

HelmCharts from jaconi's company seems a bit complicated, I think there can be a simplified version for easier understanding.

**Comment by M0NsTeRRR** [comment]
> > As I see jaconi company have shared helm chart for client and server https://artifacthub.io/packages/helm/jaconi/netbird
> > 
> > https://github.com/jaconi-io/helm-charts/tree/main
> 
> HelmCharts from jaconi's company seems a bit complicated, I think there can be a simplified version for easier understanding.

It's a classic helm chart with a lot of configurable features (all ? I didn't look much into in for now) with default values also. If we have a simple helm chart if your requirement change and need a feature that's not managed you have to migrate to another helm chart. Documentation and default values (when possible) will helps to understand.

**Comment by marcportabellaclotet-mt** [comment]
I have written a generic helm chart, which is not as coupled as the one from jaconi. Right now works quite well, and is quite configurable.
The difficult part, is how to manage the management configuration file, and populate it with configs and secrets.
Should I create a PR on this repo, with a proposal?
Spoiler alert:
- It will support configuration template with secrets and config support using [vals](https://github.com/helmfile/vals) in an init container.

**Comment by dfry** [comment]
@marcportabellaclotet-mt can you share a preview of your chart with the proposal?  thanks

**Comment by marcportabellaclotet-mt** [comment]
[Here](https://github.com/totmicro/helms/tree/main/charts) is a basic version of the Helm chart that I am using.

It needs documentation, which could be added if the proposal is accepted.
The aim is to support any configuration since it's impractical to template all possible configurations based on the management.json file.

To render the configuration, we will need to run the configure.sh from [Netbird](https://github.com/netbirdio/netbird/tree/main/infrastructure_files), include it as values.configuration in the netbird chart, and replace sensitive data with [vals](https://github.com/helmfile/vals) references. Note that netbird-dashboard also needs to update some env vars.

Maybe I could merge both charts for simplicity.

I provide an example using Authentik as an IdP and Kubernetes secrets to manage sensitive data.

For deploying the TURN server, I have successfully done so in Kubernetes using the [Stunner Gateway Operator](https://github.com/l7mp/stunner-gateway-operator).

**Comment by marcportabellaclotet-mt** [comment]
I just updated the helm chart, with relay support.
It includes examples, documentation, and now you can use it a helm release:
``` 
helm repo add netbird-helm https://totmicro.github.io/helms
```
Details [here](https://github.com/totmicro/helms).

**Comment by dfry** [comment]
@marcportabellaclotet-mt that is great work.  I had forked your repo earlier and added some elements that I needed: https://github.com/mojaloop/netbird-charts

i would like to move to using your chart repo if we can see if my changes are needed anymore and if so, make a PR from mine to yours.   

But also, was wondering if you are on the netbird slack.  There are other users there interested in a helm chart and it would be good to get the discussion going with them as well as the netbird guys.

one big thing would be to address the use of env vars in the management.json config so we don't need the helmfile secrets injection and the kubectl init container bit.

**Comment by marcportabellaclotet-mt** [comment]
I will join netbird slack, to discuss about this.

**Comment by KlavsKlavsen** [comment]
so the concensus is that you are moving forward with @marcportabellaclotet-mt 's helm chart.. and eventually merge that into netbird repo here as the official helm chart, or ?

**Comment by drewhemm** [comment]
I am also happy to contribute to this effort.

**Comment by dfry** [comment]
join the netbird slack #helm channel

**Comment by MohammedNoureldin** [comment]
It is not an issue, we are missing the official Helm, while I am currently using my self-written chart. FYI @nazarewk

**Comment by richevanscybermyte** [comment]
More of a feature request and not an issue but regardless, it is a need.

**Comment by hobbit44** [comment]
Can agree this is a need. Im surprised you went about creating an [operator](https://github.com/netbirdio/kubernetes-operator) before even creating a chart to host the server

Edit: Although i have just spotted [this](https://github.com/netbirdio/helms/tree/main/charts/netbird), which i will be investigating.

**Comment by mlsmaycon** [comment]
The chart is available and it was donated by the community. If you have any feedback, please join our slack workspace [here](https://docs.netbird.io/slack-url) or open an issue in the chart repository directly.

**Comment by KlavsKlavsen** [comment]
@mlsmaycon Sounds great. Where can I find it? I can't see it in general readme in this repo under installation methods?

**Comment by richevanscybermyte** [comment]
It looks like this chart (https://github.com/netbirdio/kubernetes-operator) is for the agent, I was expecting the manager and UI. Unless I am missing something.

**Comment by antoniolago** [comment]
Isn't this one? https://github.com/netbirdio/helms

**Comment by richevanscybermyte** [comment]
Yes, you are right, I was looking at the operator that was posted earlier. Thank you!

**Comment by ashish1099** [comment]
what chart are we using for coturn ?

**Comment by MohammedNoureldin** [comment]
@ashish1099 checking the source code of the Chart shows that the default TURN server is a netbird-built TURN server. You still can use your external self-hosted COTURN.

**Comment by cclloyd** [comment]
I also happened to write a helm chart (the existing one didn't fit all the needs of my cluster).  Feel free to steal anything from it https://github.com/cclloyd/helm-netbird

**Comment by ashish1099** [comment]
@MohammedNoureldin I couldn't found anything where it says netbird-built TURN server setup, in the compose file too I saw its running coturn only.

I have working here [kubeaid netbird](https://github.com/Obmondo/KubeAid/tree/master/argocd-helm-charts/netbird)
